lib/cc/analyzer/engines_builder.rb

```diff
@@ -44,7 +44,7 @@ def engine_config(raw_engine_config)
 
       def names_and_raw_engine_configs
         {}.tap do |ret|
-          @config.engines.each do |name, raw_engine_config|
+          (@config.engines || []).each do |name, raw_engine_config|
```

Comment:
Kind of. Because of how CC::YAML works, if the engines value is invalid, it won't actually be present in the fully parsed node. An empty engines node is considered invalid now.

I was going to suggest that CLI commands that use config values should probably run `validate-config` before they do anything else. But I think that should be a separate PR. In this particular case this is just a change to maintain existing behavior due to changing dependency behavior.

> On Nov 4, 2015, at 08:01, Francis Hwang <notifications@github.com> wrote:
> 
> In lib/cc/analyzer/engines_builder.rb:
> 
> > @@ -44,7 +44,7 @@ def engine_config(raw_engine_config)
> >  
> >        def names_and_raw_engine_configs
> >          {}.tap do |ret|
> > -          @config.engines.each do |name, raw_engine_config|
> > +          (@config.engines || []).each do |name, raw_engine_config|
> Out of curiosity, is this for the case where the engines array is entirely missing from the YAML?
> 
> —
> Reply to this email directly or view it on GitHub.
>